src/util/console.cpp

```diff
@@ -3,96 +3,145 @@
-            qWarning() << "Could not get stderr handle. Error code:" << GetLastError();
         }
 
         // Save current code page
         m_oldCodePage = GetConsoleOutputCP();
         m_shouldResetCodePage = true;
 
+        // Verify using the unicode font "Consolas"
         HMODULE kernel32_dll = LoadLibraryW(L"kernel32.dll");
+        HANDLE hStdOut = GetStdHandle(STD_OUTPUT_HANDLE);
         if (kernel32_dll && hStdOut != INVALID_HANDLE_VALUE) {
-            pfSetCurrentConsoleFontEx pfSCCFX = (pfSetCurrentConsoleFontEx)GetProcAddress(kernel32_dll, "SetCurrentConsoleFontEx");
-            if (pfSCCFX) {
-                // Use a unicode font
-                CONSOLE_FONT_INFOEX newFont;
-                newFont.cbSize = sizeof newFont;
-                newFont.nFont = 0;
-                newFont.dwFontSize.X = 0;
-                newFont.dwFontSize.Y = 14;
-                newFont.FontFamily = FF_DONTCARE;
-                newFont.FontWeight = FW_NORMAL;
-                wcscpy_s(newFont.FaceName, L"Consolas");
-                pfSCCFX(hStdOut, FALSE, &newFont);
-            } else {
-                // This happens on Windows XP
-                qWarning() << "The console font may not support non ANSI characters." <<
-                              "In case of character issues switch to font \"Lucida Console\"";
+            pfGetCurrentConsoleFontEx pfGCCFX =
+                    (pfGetCurrentConsoleFontEx) GetProcAddress(kernel32_dll,
+                            "GetCurrentConsoleFontEx"); // Supported from Windows Vista
+            pfSetCurrentConsoleFontEx pfSCCFX =
+                    (pfSetCurrentConsoleFontEx) GetProcAddress(kernel32_dll,
+                            "SetCurrentConsoleFontEx"); // Supported from Windows Vista
+            bool setFont = true;
+            if (pfGCCFX) {
+                CONSOLE_FONT_INFOEX font;
+                if (pfGCCFX(hStdOut, FALSE, &font)) {
+                    if (wcsncmp(font.FaceName, L"Consolas", 8) == 0) {
+                        // Nothing to do
+                        setFont = false;
+                    }
+                }
+            }
+
+            if (setFont) {
+                if (pfSCCFX) {
+                    // Use a unicode font
+                    CONSOLE_FONT_INFOEX newFont;
+                    newFont.cbSize = sizeof newFont;
+                    newFont.nFont = 0;
+                    newFont.dwFontSize.X = 0;
+                    newFont.dwFontSize.Y = 14;
+                    newFont.FontFamily = FF_DONTCARE;
+                    newFont.FontWeight = FW_NORMAL;
+                    wcscpy_s(newFont.FaceName, L"Consolas");
+                    pfSCCFX(hStdOut, FALSE, &newFont);
+                } else {
+                    // This happens on Windows XP
+                    qWarning() << "The console font may not support non ANSI characters." <<
+                                  "In case of character issues switch to font \"Lucida Console\"";
```

Comment:
If "Lucida Console" is known to work, maybe we should only write this warning if the font is not "Lucida Console", so better check the font against "Lucida Console" in addtition to "Consolas".

Reply:
"Consolas" is not available on Windows XP.

I have already tried a bit to make this part smarter but I failed. It is to hard for me to do it on Linux using the CI only. The best solution would to not touch the user font at all If it already supports codepage 1252.

Reply:
I mean when we check if we are already using "consolas" (some lines upper), maybe we can check if we use either "consolas" OR "Lucida Console" if the two are known to work.

This would avoid telling XP users that already use "Lucida Console" that they would better use "Lucida console". This is just some minor user experience improvement.